src/components/TodoList/TodoList.tsx

```diff
@@ -1,210 +1,88 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { TodoItem } from '../TodoItem/TodoItem';
+import { useAppSelector } from '../../app/hooks';
+import { Status } from '../../types/Status';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+};
+
+function getVisibleTodos(
+  allTodos: Todo[],
+  searchQuery: string,
+  filter: Status,
+): Todo[] {
+  let filteredTodos = [...allTodos];
+
+  if (searchQuery) {
+    const lowerQuery = searchQuery.toLowerCase();
+
+    filteredTodos = filteredTodos
+      .filter(todo => todo.title.toLowerCase().includes(lowerQuery));
+  }
+
+  if (filter) {
+    switch (filter) {
+      case 'active':
+        filteredTodos = filteredTodos.filter(todo => !todo.completed);
+        break;
+
+      case 'completed':
+        filteredTodos = filteredTodos.filter(todo => todo.completed);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  return filteredTodos;
+}
+
+export const TodoList: React.FC<Props> = ({ todos }) => {
+  const query = useAppSelector(state => state.filter.query);
+  const filterByStatus = useAppSelector(state => state.filter.status);
```

Comment:
Also you could destructure the properties 
```suggestion
  const { query, status: filterByStatus } = useAppSelector(state => state.filter);
```